jerry-port/default/default-socket.c

```diff
@@ -0,0 +1,246 @@
+  }
+
+  *bytes_sent = send (connection_p->fd, data_p, data_len, 0);
+
+  if (*bytes_sent < 0)
+  {
+    if (errno == EWOULDBLOCK)
+    {
+      return JERRY_CONN_ERROR_AGAIN;
+    }
+    else
+    {
+      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s\n", strerror (errno));
+      return JERRY_CONN_ERROR_IO;
+    }
+  }
+
+  return JERRY_CONN_ERROR_NONE;
+} /* jerry_debugger_socket_send */
+
+/**
+ * Default implementation of debugger receive api.
+ * Receive message from the client side.
+ *
+ * @return JERRY_CONN_ERROR_NONE - if the data was received successfully from the client side
+ *         JERRY_CONN_ERROR_INVALID - if the connection is invalid
+ *         JERRY_CONN_ERROR_AGAIN - if there's no incoming data, you should try again later
+ *         JERRY_CONN_ERROR_IO - if the data failed to receive
+ * Note:
+ *      This function is only available if the port implementation library is
+ *      compiled with the JERRY_DEBUGGER macro.
+ */
+static jerry_debugger_conn_errors_t
+jerry_debugger_socket_receive(jerry_debugger_conn_t *connection_p, /**< connection pointer */
+                              void *data_p, /**< data pointer */
+                              size_t max_len, /**< max data size */
+                              ssize_t *bytes_received) /**< bytes received */
+{
+  if (!connection_p)
+  {
+    *bytes_received = -1;
+    return JERRY_CONN_ERROR_INVALID;
+  }
+
+  *bytes_received = recv (connection_p->fd, data_p, max_len, 0);
+
+  if (*bytes_received < 0)
+  {
+    if (errno == EWOULDBLOCK)
+    {
+      return JERRY_CONN_ERROR_AGAIN;
+    }
+    else
+    {
+      jerry_port_log (JERRY_LOG_LEVEL_ERROR, "Error: %s\n", strerror (errno));
+      return JERRY_CONN_ERROR_IO;
+    }
+  }
+
+  return JERRY_CONN_ERROR_NONE;
+} /* jerry_debugger_socket_receive */
+
+/**
+ * Default implementation of debugger close api.
+ * Closes the debugger connection.
+ *
+ * @return JERRY_CONN_ERROR_NONE - if successful
+ *         JERRY_CONN_ERROR_INVALID - if the connection is invalid
+ * Note:
+ *      This function is only available if the port implementation library is
+ *      compiled with the JERRY_DEBUGGER macro.
+ */
+static jerry_debugger_conn_errors_t
+jerry_debugger_socket_close (jerry_debugger_conn_t *connection_p) /**< connection pointer */
+{
+  if (!connection_p)
+  {
+    return JERRY_CONN_ERROR_INVALID;
+  }
+
+  close (connection_p->fd);
+  return JERRY_CONN_ERROR_NONE;
+} /* jerry_debugger_socket_close */
+
+static jerry_debugger_transport_t socket_transport = {
+  .config_p = &socket_config,
+  .accept  = jerry_debugger_socket_accept,
+  .send = jerry_debugger_socket_send,
+  .receive = jerry_debugger_socket_receive,
+  .close = jerry_debugger_socket_close,
+};
+
+/**
+ * Create and return the socket transport on the provided port for the debugger
+ *
+ * @return the transport created
+ */
+jerry_debugger_transport_t*
+jerry_debugger_init_socket_transport(uint16_t tcp_port) /**< server port number */
+{
```

Comment:
Perhaps add an assert to make sure it's only called once (since the `socket_transport` state is global/singleton).

Reply:
Hmm I see, ok, I guess it's ok.

There's nothing precluding one from creating a dynamically created transport (and having a transport-specific cleanup function to free/destroy the transport), so I'm happy.